### DISCUSSION TITLE
element state (eg, button) in custom code

### DISCUSSION THREAD

**Discussion by flexiti** [HASwitchPlate/openHASP#653] [Q&A] [answered]
What is the best method to locally check the status of e.g. a buttonis on or off  in custom code?

**Comment by fvanroie** [comment, marked as answer]
First you need to get the object pointer:

```cpp
    uint8_t page = 1;   // the page of the object you want
    uint8_t id   = 150; // the id of the object you want

    lv_obj_t* my_obj = hasp_find_obj_from_page_id(page, id);
    if(!my_obj ) return; // object doesn't exist
```

Then you can use the object pointer to get the state, for example: `lv_btn_get_state(my_obj)` which will return the [current state](https://docs.lvgl.io/7.11/widgets/btn.html#states).

```cpp
   lv_state_t  val = lv_btn_get_state(my_obj );
   if (val & LV_STATE_CHECKED) {
        // do something
   } else {
        // do something
   }
```

**Comment by flexiti** [comment]
Perfect, and it even works ;) Thank you. ps. Can other resources be read in a similar way? e.g. current text from the label - received value from mqtt that is not a castom message

**Reply by fvanroie** [reply]
Yes, you can use all LVGL apis in this way.
